### DISCUSSION TITLE
TCP handshake server not bounded to 9999

### DISCUSSION THREAD

**Discussion by 03psb** [ANLAB-KAIST/KENSv3#217] [Project Q&A 2. Transport Layer (TCP)] [answered]
Hi, I'm working on TCP handshake testcase. In backlog1 test, the server socket is bound to 9999 and perform handshake with clients.
<img width="659" alt="캡처" src="https://github.com/ANLAB-KAIST/KENSv3/assets/98043513/7684d314-ffc6-468a-9a4a-5d35bb3b5f99">
but when I actually ran the testcase, I found server socket is bound to 3879, not 9999 and creating problem. Is bound to 3879 unintended and indicates some fault in my implementation? Or is it implies there needs to be some redirection mechanism that resides on 9999 and sets up a connection with 3879?

**Comment by Pusnow** [comment, marked as answer]
It seems that you do not handle endian correctly.

`3879` is network order of `9999`, that is
```c
3879 == htons(9999)
```

Please check ordering again.

**Reply by 03psb** [reply]
I found out it was indeed endian problem. Thank you for your answer
